apis/v1alpha1/rule_groups_namespace.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RuleGroupsNamespaceSpec defines the desired state of RuleGroupsNamespace.
+type RuleGroupsNamespaceSpec struct {
+	// The namespace data that define the rule groups.
+	Data []byte `json:"data,omitempty"`
```

Comment:
If i understand well users will have to provide one of `Data` or `Configuration`?

Reply:
I just realized `Data` is never used. Is possible to remove this field but ignoring the field shape?

Reply:
I don't think its possible to remove the `Data` field. There is some places where the generated code using the `Data` field is necessary to have. For example in `newCreateRequestPayload`, its not possible to insert a hook there to set the API Input data field to something other than `r.ko.Spec.Data`.  So before calling `newCreateRequestPayload`, I have to set that `Data` field to the byte slice.

Reply:
If we remove the `Data` field, the code-generator will not generate any code that will try to use it.

Reply:
You are right, I was forgetting about a particular other hook. The data field is now completely removed.